src/app/components/login/login.component.ts

```diff
@@ -1,64 +1,82 @@
-import { AuthService } from './../../services/auth.service';
+import { AuthResponse } from './../../auth-response';
 import { CommonModule } from '@angular/common';
-import { Component } from '@angular/core';
-import { FormControl, FormGroup, ReactiveFormsModule, Validators, FormBuilder } from '@angular/forms';
+import { Component, OnInit } from '@angular/core';
+import {
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+  FormBuilder,
+} from '@angular/forms';
 import { LoginRequest } from '../../login-request';
 import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { selectToken } from '../store/selectors/login.selectors';
+import { loginPage } from '../store/actions/login.actions';
+import { filter, first } from 'rxjs';
 
 @Component({
   selector: 'app-login',
   standalone: true,
   imports: [ReactiveFormsModule, CommonModule],
   templateUrl: './login.component.html',
-  styleUrl: './login.component.scss'
+  styleUrl: './login.component.scss',
 })
-export class LoginComponent {
-  email : FormControl<string | null> = new FormControl<string>('', [Validators.required, Validators.email]) 
-  password : FormControl<string | null> = new FormControl<string>('', [Validators.required, Validators.minLength(5)]) 
-  
+export class LoginComponent implements OnInit {
+  email: FormControl<string | null> = new FormControl<string>('', [
+    Validators.required,
+    Validators.email,
+  ]);
+  password: FormControl<string | null> = new FormControl<string>('', [
+    Validators.required,
+    Validators.minLength(5),
+  ]);
+
   loginForm!: FormGroup;
 
-  inlineNotification: {show: boolean; type: string; text: string} = {
+  inlineNotification: { show: boolean; type: string; text: string } = {
     show: false,
     type: '',
-    text: ''
-  }
-  constructor(private FormBuilder: FormBuilder, private authService: AuthService, private router: Router) {
-     this.loginForm = this.FormBuilder.group({
+    text: '',
+  };
+  constructor(
+    private FormBuilder: FormBuilder,
+    private router: Router,
+    private store: Store
+  ) {
+    this.loginForm = this.FormBuilder.group({
       email: this.email,
-      password: this.password
-     })
+      password: this.password,
+    });
+  }
+
+  ngOnInit(): void {
+    this.store
```

Comment:
You are still doing this. This is the work of guards

Reply:
This has not been resolved

Reply:
@Bilal-Abubakari Please It has been resolved pushing the component code now